### DISCUSSION TITLE
Rp2040 Pico- best codec and setup

### DISCUSSION THREAD

**Discussion by SnazzyBlanket** [pschatzmann/arduino-audio-driver#14] [Q&A] [answered]
I apologize if this has been covered. I am trying to decide which I2S codec to purchase to use with the PICO so that it will work properly with the arduino-audio driver or arduino audio tools repo (With the ultimate goal of running Arduino STK on the pico)

I was not sure  since the rp2040 does not have a built in I2s and you need to use the PIo Machines to create I2s. thanks for all the work you do!

**Comment by pschatzmann** [comment, marked as answer]
I am not sure if this is a good combination: STK uses a lot of floating point operations and the RP2040 is pretty bad with this, so you might need to adjust (lower) the sampling rate to get some acceptable audio.

Apart from this, any [DAC described in the Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-DAC) will do!

ps. you don't need this project for this, but rather only the AudioTools!

**Reply by SnazzyBlanket** [reply]
ahh Ok i see--thanks...I could definitely try lowering the sampling rate.

if you were starting an audio project from scratch which processor would you recommend? is it possible the ESP32 chips are better speced than the pico? I suppose the advantage with the pico is price

thanks

**Comment by SnazzyBlanket** [comment]
i am going to order a HiLetgo PCM5102 I2S IIS today!

**Comment by pschatzmann** [comment]
The ESP32 has more RAM. I recommend to get a version with PSRAM.
The floating point operations are about 3 times faster.
And the Bluetooth and Wifi functionality makes it possible to send and receive audio.

I think the prices start around 3 USD, so it is also pretty cheap...

**Comment by SnazzyBlanket** [comment]
Ok thanks--- I did some searching in my parts bins and found an old Huzza32 someone gave me years back so i will start by testing that. (it also says it has two built in 8 bit DAC) and I will look into ordering one with PSRAM

**Comment by pschatzmann** [comment]
I got lazy and therefore prefer to work with boards like the LyraT or the AI Thinker AudioKit
